test/hw8/test_task1.py

```diff
@@ -0,0 +1,95 @@
+from tempfile import NamedTemporaryFile
+from unittest.mock import patch
+
+import pytest
+
+from All_home_works.hw8.task1 import KeyValueStorage
+
+
+@pytest.fixture(scope="module")
+def file_name():
+    file = NamedTemporaryFile(mode="w")
+    data = ["name=NAME\n", "new_1=NEW_1\n", "int_value=10\n"]
+    file.writelines(data)
+    file.seek(0)
+    yield file.name
+    file.close()
+
+
+def test_key_value_storage__has_attributes(file_name):
+    storage = KeyValueStorage(file_name)
+    assert storage.name == "NAME"
+    assert storage.new_1 == "NEW_1"
+    assert storage.int_value == 10
+
+
+def test_key_value_storage__has_attributes_as_item(file_name):
+    storage = KeyValueStorage(file_name)
+    assert storage["name"] == "NAME"
+    assert storage["new_1"] == "NEW_1"
+    assert storage["int_value"] == 10
+
+
+def test_key_value_storage__not_existing_attributes(file_name):
+    storage = KeyValueStorage(file_name)
+    with pytest.raises(AttributeError, match="object has no attribute"):
+        error = storage.not_existing
+    with pytest.raises(AttributeError, match="object has no attribute"):
+        error = storage["not_existing"]  # noqa: F841
```

Comment:
What is the recommended exce[tion in that case?

Reply:
ValueError

Reply:
Could a provide a link to the explanation?

Reply:
> In case when value cannot be assigned to an attribute
(for example when there's a line 1=something) ValueError should be raised.

So let cut brackets.
> In case when value cannot be assigned to an attribute ValueError should be raised.

And in my case, I couldn't assign value, because I do not have such an attribute. So in some sense, it should be a ValueError exception.

Reply:
I am not sure what is `value` and what is `attribute` here.

Reply:
I'm a little bit confusing. But if read the description with brackets for me it's better to use AttributeError, because we do not have such an attribute. But in the same way, it can be a ValueError because of description. So what do you think?

Reply:
> description with brackets 

That thing has a name. 

> for me it's better to use AttributeError, because we do not have such an attribute

Please dig a bit into documentation and describe which syntax in Python leads to AttributeError.

> it can be a ValueError because of description.

Do you mean that because we should raise ValueError when we construct our class we also should raise it when we use our class?